consensus/src/chained_bft/event_processor.rs

```diff
@@ -820,26 +814,51 @@ impl<T: Payload> EventProcessor<T> {
                 return;
             }
         };
-        // At this moment the new state is persisted and we can notify the clients.
-        // Multiple blocks might be committed at once: notify about all the transactions in the
-        // path from the old root to the new root.
-        for committed in blocks_to_commit {
-            if let Some(time_to_commit) = duration_since_epoch()
-                .checked_sub(Duration::from_micros(committed.timestamp_usecs()))
+
+        // update counters
+        for block in blocks_to_commit {
+            if let Some(time_to_commit) =
+                duration_since_epoch().checked_sub(Duration::from_micros(block.timestamp_usecs()))
             {
                 counters::CREATION_TO_COMMIT_S.observe_duration(time_to_commit);
             }
-            if let Some(payload) = committed.payload() {
-                let compute_result = committed.compute_result();
-                if let Err(e) = self
-                    .txn_manager
-                    .commit_txns(payload, &compute_result, committed.timestamp_usecs())
-                    .await
-                {
-                    error!("Failed to notify mempool: {:?}", e);
+            counters::COMMITTED_BLOCKS_COUNT.inc();
+            let block_txns = block.output().transaction_data();
+            counters::NUM_TXNS_PER_BLOCK.observe(block_txns.len() as f64);
+
+            for txn in block_txns.iter() {
+                match txn.txn_info_hash() {
+                    Some(_) => {
+                        counters::COMMITTED_TXNS_COUNT
+                            .with_label_values(&["success"])
+                            .inc();
+                    }
+                    None => {
+                        counters::COMMITTED_TXNS_COUNT
+                            .with_label_values(&["failed"])
+                            .inc();
+                    }
+                }
+            }
+            for status in block.compute_result().compute_status.iter() {
+                match status {
+                    TransactionStatus::Keep(_) => {
+                        counters::COMMITTED_TXNS_COUNT
+                            .with_label_values(&["success"])
+                            .inc();
+                    }
+                    TransactionStatus::Discard(_) => {
+                        counters::COMMITTED_TXNS_COUNT
+                            .with_label_values(&["failed"])
+                            .inc();
+                    }
                 }
             }
         }
+
+        // At this moment the new state is persisted and we can notify the clients.
```

Comment:
These comments are not relevant anymore: the client notification happens earlier when you're calling `block_store.commit()`, which eventually notifies state synchronizer, which is now notifying the clients.